sdk/protocol-plugins/src/implementation/BaseProtocolPlugin.ts

```diff
@@ -136,8 +136,14 @@ export abstract class BaseProtocolPlugin implements IProtocolPlugin {
   /** ACTION BUILDERS */
 
   /** @see IProtocolPlugin.getActionBuilder */
-  getActionBuilder<T extends steps.Steps>(step: T): Maybe<ActionBuilder<T>> {
-    return this.stepBuilders[step.type] as ActionBuilder<T>
+  getActionBuilder<StepType extends steps.Steps>(step: StepType): Maybe<IActionBuilder<StepType>> {
+    const builder = this.stepBuilders[step.type]
+
+    if (!builder) {
+      return undefined
+    }
+
+    return new builder() as IActionBuilder<StepType>
   }
```

Comment:
consider renaming `builder` to `Builder` - not required as it's a variable, but might be easier to read to signal it's a variable holding a reference to a class

Reply:
Agree, I'll change it to BuilderClass